app/data_providers/dispatch.py

```diff
@@ -66,11 +67,56 @@ async def get_ranked_corpus_slate(self, user: UserIds, recommendation_count: int
             recommendations=recommendations,
         )
 
-        await self.slate_tracker.track(corpus_slate, user=user)
-
         return corpus_slate
 
 
+class HomeDispatch:
+
+    def __init__(
+            self,
+            corpus_client: CorpusFeatureGroupClient,
+            user_recommendation_preferences_provider: UserRecommendationPreferencesProvider,
+            topic_provider: TopicProvider,
+            topic_slate_provider: TopicSlateProvider,
+    ):
+        self.topic_provider = topic_provider
+        self.corpus_client = corpus_client
+        self.user_recommendation_preferences_provider = user_recommendation_preferences_provider
+        self.topic_slate_provider = topic_slate_provider
+
+        self.setup_moment_dispatch = self._create_setup_moment_dispatch()
+
+    async def get_slate_lineup(
+            self, user: UserIds, slate_count: int, recommendation_count: int
+    ) -> CorpusSlateLineupModel:
+        slates = [await self.setup_moment_dispatch.get_ranked_corpus_slate(
```

Comment:
we should always use `asyncio.gather` when we can call several async functions at the same time

Reply:
This is a single function call. If clearer, I could split this into two lines:
```suggestion
        slates = []
        slates.append(await self.setup_moment_dispatch.get_ranked_corpus_slate(
```

Reply:
we have another async function on the next line that potentially does IO operations to retrieve topics